runtime/near-vm-logic/src/logic.rs

```diff
@@ -2786,6 +2786,67 @@ impl<'a> VMLogic<'a> {
         }
         Ok(())
     }
+
+    /// Verify an ED25519 signature given a message and a public key.
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + ed25519_verify_base + ed25519_verify_byte * num_bytes`
+
+    pub fn ed25519_verify(
+        &mut self,
+        sig_len: u64,
+        sig_ptr: u64,
+        msg_len: u64,
+        msg_ptr: u64,
+        pub_key_len: u64,
+        pub_key_ptr: u64,
+        register_id: u64,
+    ) -> Result<()> {
+        use ed25519_dalek::{PublicKey, Signature, Verifier};
+
+        self.gas_counter.pay_base(ed25519_verify_base)?;
+        self.gas_counter.pay_per(ed25519_verify_byte, msg_len as u64)?;
```

Comment:
Doesn't seem like we have charged all the fees here.  write register related fees are missing.